x-pack/plugins/apm/public/components/shared/time_comparison/index.tsx

```diff
@@ -41,13 +41,13 @@ export function TimeComparison() {
   const { start, end } = useTimeRange({ rangeFrom, rangeTo });
 
   const {
-    urlParams: { comparisonEnabled, comparisonType },
+    urlParams: { comparisonEnabled, offset },
```

Comment:
any reason why we're still getting this from `useLegacyUrlParams()`?

Reply:
We've discussed this previously here https://github.com/elastic/kibana/pull/126567#discussion_r822801743. Basically if we get it from apm params it'll be false the first time instead of undefined and we wouldn't know if it was actually in the url or we should query the advanced settings. Will the redirect you've suggested overcome this issue?

Reply:
Are you referring to `comparisonEnabled` or `offset`? I was referring to the latter. I'm hoping we can make `comparisonEnabled` required and redirect in `RedirectWithOffset` when taking the value of the Kibana Advanced Setting into acocunt - would that make sense?

Reply:
I was referring to `comparisonEnabled`. Removed the call to `useLegacyUrlParams`, made `comparisonEnabled` required and added redirect in https://github.com/elastic/kibana/pull/127812/commits/f0c8d72ef433c74b468617d6f0fb144d7af65f82.